### PR TITLE
test: add latest/earliest available attestation nonce keys tests

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/celestia-app/issues/1908

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: x/qgb/keeper/keeper_attestation_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	k := input.QgbKeeper
+
+	// check if the earliest available attestation nonce is init
+	exists := k.CheckEarliestAvailableAttestationNonce(input.Context)
+	assert.False(t, exists)
+
+	// init the earliest available attestation nonce
+	input.QgbKeeper.SetEarliestAvailableAttestationNonce(input.Context, qgb.InitialEarliestAvailableAttestationNonce)
+
+	// check if the latest attestation nonce value was initialized correctly
```
Comment (rootulp) [inline x/qgb/keeper/keeper_attestation_test.go:38]:
```suggestion
	// check if the earliest attestation nonce value was initialized correctly
```

File: x/qgb/keeper/keeper_valset_test.go [hunk 2]
```diff
@@ -65,8 +67,59 @@ func TestCurrentValsetNormalization(t *testing.T) {
 	}
 }
 
-func TestCheckingLatestAttestationNonceInValsets(t *testing.T) {
-	input := testutil.CreateTestEnvWithoutAttestationNonceInit(t)
+func TestCheckingEarliestAvailableAttestationNonceInValsets(t *testing.T) {
+	input := testutil.CreateTestEnvWithoutQGBKeysInit(t)
+	k := input.QgbKeeper
+	// create a validator to have a  realistic scenario
```
Comment (rootulp) [inline x/qgb/keeper/keeper_valset_test.go:73]:
```suggestion
	// create a validator to have a realistic scenario
```

**Review by rootulp — DISMISSED** [review]

File: x/qgb/keeper/keeper_data_commitment_test.go [hunk 3]
```diff
@@ -186,3 +188,48 @@ func TestCheckingLatestAttestationNonceInDataCommitments(t *testing.T) {
...
+		},
+		{
+			name: "check earliest available attestation nonce before getting latest data commitment",
+			requestFunc: func() error {
+				_, err := k.GetLatestDataCommitment(input.Context)
+				return err
+			},
+			expectedError: types.ErrEarliestAvailableNonceStillNotInitialized,
+		},
+	}
```
Comment (evan-forbes) [inline x/qgb/keeper/keeper_data_commitment_test.go:228]: do we have these, and the other tests besides TestCheckLatestAttestationNonce, just to ensure that we are calling `CheckLatestAttestationNonce` in those functions?

**Review by evan-forbes — APPROVED** [review]
`TestCheckLatestAttestationNonce` is really clear and straight forward!

certainly not going to block on that though :slightly_smiling_face: LGTM

File: x/qgb/keeper/keeper_data_commitment_test.go (hunk 3 above)
Comment (SweeXordious) [reply x/qgb/keeper/keeper_data_commitment_test.go:228]: If those functions were not calling the check method, that error will not be returned. The main reason we're checking for that specific error is to see if the check is called.

Is this what you meant? Or im missing your question

Comment (evan-forbes) [reply x/qgb/keeper/keeper_data_commitment_test.go:228]: yeah that makes sense. I was just making sure that I understood the purpose of the tests.

File: x/qgb/genesis.go [hunk 4]
```diff
@@ -6,15 +6,24 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+const (
+	// InitialLatestAttestationNonce the initial value set in genesis of the latest attestation
+	// nonce value in store.
+	InitialLatestAttestationNonce = uint64(0)
+	// InitialEarliestAvailableAttestationNonce the initial value set in genesis of the earliest
+	/// available attestation nonce in store.
+	InitialEarliestAvailableAttestationNonce = uint64(1)
```
[the remark this answers is not in the document]
Comment (SweeXordious) [reply x/qgb/genesis.go:15]: There is a comment explaining why... It's mostly a gas optimization not to have to check the genesis case everytime in end block

**Pull request merged**